### PR TITLE
Some Ux issues from mobile testing

### LINKED ISSUE #1123: Ux Misc
- [x] View stock: cannot sort by code or name
- [x] View stock: search is incorrectly parsing values as numeric
- [x] Items: Add search bar
- [x] Remove `Confirmed` column on inbound shipment list view
- [x] OS: When looking at the outbound shipment the table doesn't fit in the tablet and no column can be deleted 
- [x] The "hold" button does not work for new 'inbound shipments' that have not been finalized

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #1123 

Changes in this PR

- [ ] View stock: can sort by item code, item name, batch, pack size, total quantity and supplier name
- [ ] View stock: search is now handling numeric search entries
- [ ] Items list: Added search bar
- [ ] Items list: Added soh, amc and mos columns ( yes, not requested. but the list looked boring. )
- [ ] Items list: soh value rounded to an int, amc and mos are 2 decimal places
- [ ] Inbound list: Replaced `Confirmed` column with 'Delivery Date'
- [ ] Inbound detail: Enabled the hold button functionality
- [ ] Outbound list: Enabled column selection for the table
- [ ] Filter input now has a 'clear filter' option; when a value has been entered you can click the little `x` and clear the filter term

Screenshot 2023-02-08 at 12 24 54 PM

Screenshot 2023-02-08 at 12 37 51 PM

### CODE REVIEW

File: client/packages/common/src/hooks/useUrlQuery/useUrlQueryParams.ts [hunk 1]
```diff
@@ -22,7 +22,9 @@ export const useUrlQueryParams = ({
   // do not coerce the filter parameter if the user enters a numeric value
   // if this is parsed as numeric, the query param changes filter=0300 to filter=300
   // which then does not match against codes, as the filter is usually a 'startsWith'
-  const { urlQuery, updateQuery } = useUrlQuery({ skipParse: ['filter'] });
+  const { urlQuery, updateQuery } = useUrlQuery({
+    skipParse: [filterKey ?? 'filter'],
```
Comment (mark-prins) [inline client/packages/common/src/hooks/useUrlQuery/useUrlQueryParams.ts:26]: prevent the `useUrlQuery` from parsing the filter values as numbers or dates. strings only for filters.

File: client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/hooks/useNextItem.test.tsx [hunk 2]
```diff
@@ -80,6 +80,7 @@ const getLines = (): InvoiceLineNode[] => [
         averageMonthlyConsumption: 0,
       },
       atcCategory: '',
+      availableStockOnHand: 0,
```
Comment (mark-prins) [inline client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/hooks/useNextItem.test.tsx:83]: required when I generated the schema - comes from recent changes to the api

File: server/repository/src/db_diesel/stock_line.rs [hunk 3]
```diff
@@ -121,11 +146,15 @@ impl<'a> StockLineRepository<'a> {
     }
 }
 
-type BoxedStockLineQuery =
-    IntoBoxed<'static, LeftJoin<LeftJoin<stock_line::table, location::table>, name::table>, DBType>;
+type BoxedStockLineQuery = IntoBoxed<
+    'static,
+    LeftJoin<LeftJoin<InnerJoin<stock_line::table, item::table>, location::table>, name::table>,
```
Comment (mark-prins) [inline server/repository/src/db_diesel/stock_line.rs:151]: I couldn't see a way to order the results by the result of the item loader - so I've just joined the item table to the stock line 🤷

**Review by CarlosNZ — APPROVED** [review]
Front-end LGTM! Maybe someone else wants to have a quick look over the back-end?

**Pull request merged**